frontend/packages/dev-console/src/components/catalog/catalog-view/CatalogView.tsx

```diff
@@ -215,44 +241,15 @@ const CatalogView: React.FC<CatalogViewProps> = ({
   }, [activeGrouping, filteredItems, isGrouped]);
 
   React.useEffect(() => {
-    let unmounted = false;
-    const searchParams = new URLSearchParams(window.location.search);
-    const categoryParam = searchParams.get('category');
-    const keywordParam = searchParams.get('keyword');
-    const groupingParam = searchParams.get('grouping');
-    const attributeFilters = {};
-
-    _.each(filterGroups, (filterGroup) => {
-      const attributeFilterParam = searchParams.get(filterGroup);
-      try {
-        _.set(attributeFilters, filterGroup, JSON.parse(attributeFilterParam));
-      } catch (e) {
-        // eslint-disable-next-line no-console
-        console.warn('could not update filters from url params: could not parse search params', e);
-      }
-    });
-
-    const newFilters = getActiveFilters(
-      attributeFilters,
-      availableFilters,
-      filterStoreKey,
-      filterRetentionPreference,
-    );
-
-    if (!unmounted) {
-      categoryParam && setActiveCategoryId(categoryParam);
-      keywordParam && setActiveSearchKeyword(keywordParam);
-      groupingParam && setActiveGrouping(groupingParam);
-      newFilters && setActiveFilters(newFilters);
-    }
-
-    // this doesn't work right now because of issue with PF SearchInput
-    searchInputRef.current && searchInputRef.current.focus({ preventScroll: true });
-
-    return () => {
-      unmounted = true;
-    };
-  }, [availableFilters, filterGroups, filterRetentionPreference, filterStoreKey]);
+    // TODO: Remove this hack once https://github.com/patternfly/patternfly-react/issues/5168 is fixed.
+    // eslint-disable-next-line react/no-find-dom-node
+    const toolbarDOMNode = ReactDOM.findDOMNode(catalogToolbarRef.current) as HTMLDivElement;
+    searchInputRef.current = toolbarDOMNode.querySelector(
+      '.pf-c-search-input__text-input',
+    ) as HTMLInputElement;
+
+    searchInputRef && searchInputRef.current.focus({ preventScroll: true });
```

Comment:
`CatalogView` shouldn't care about the search input. It should call `focus()` directly on the toolbar ref.
Leave it up to the `CatalogToolbar` to delegate focus to the SearchInput.

Reply:
Done.